spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -136,10 +136,12 @@ protected Object doReceive() {
 			String remotePath = remotePath(file);
 			Session<?> session = this.remoteFileTemplate.getSession();
 			try {
-				return getMessageBuilderFactory().withPayload(session.readRaw(remotePath))
+				return getMessageBuilderFactory()
+						.withPayload(session.readRaw(remotePath))
 						.setHeader(IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE, session)
 						.setHeader(FileHeaders.REMOTE_DIRECTORY, file.getRemoteDirectory())
 						.setHeader(FileHeaders.REMOTE_FILE, file.getFilename())
+						.setHeader(FileHeaders.REMOTE_FILE_INFO, (file.toJson()))
```

Comment:
So, you still aren't going to provide alternative with direct object, not JSON. Are you?